scripts/compiledDefinesTemplate.hpp.in

```diff
@@ -28,6 +28,7 @@
 #cmakedefine01 OCCA_HIP_ENABLED
 #cmakedefine01 OCCA_METAL_ENABLED
 
+#cmakedefine OCCA_SOURCE_DIR "@OCCA_SOURCE_DIR@"
```

Comment:
Why do we need a `OCCA_BUILD_DIR` and `OCCA_SOURCE_DIR`?

Reply:
Ya, this is the sticky part. Typically a project it going to create some build folder somewhere and then point to occa as a dependency. The default behavior in cmake is then to compile and put libocca.so in the build/lib folder, whose path could be completely different to the actual location of occa. Hence, there's a need to keep two directory locations, one to find occa/include and one to find libocca.so. 

The way I see it we could do one of the following things:

1. Install libocca.so into occa/lib like the GNU make system does. This isn't typical of cmake build systems since it's often assumed that removing the entire build folder itself removes all the build-related objects (but we're already breaking that rule somewhat with compiledDefines.hpp).
2. Copy the entire contents of occa/include to build/include. The old cmake build system was doing this. Needless to say, I don' t think this is the right thing to do, but it eliminates the need to keep two directory locations.
3. Carry around the two separate BUILD and SOURCE directories. 

I'm open to anything. 1 is the closest to the current make system behavior and doesn't require src changes.

Reply:
Ah that makes sense. In that case I think this is the right approach to properly handle both builds. 

Can we save the build dir as env::OCCA_INSTALL_DIR (open to renaming this) which removed the slash? If cmake doesn't intent to move the .so maybe we can keep it named as env::OCCA_BUILD_DIR

Thank you for the detailed explanation + alternatives!

Reply:
Sorry for being a bit late to the party here--but for projects using CMake, if the `libocca` include directories are set using `target_include_directories`, then anything with an upstream dependency on OCCA will inherit the correct include flags when they call `target_link_libraries`. 

In the case of compiling CUDA or SERIAL kernels, this won't help; however, I think that the naming convention `OCCA_BUILD_DIR` and `OCCA_SOURCE_DIR` is possibly confusing when what is really meant is something like `OCCA_LIBRARY_DIR` and `OCCA_INCLUDE_DIR`. If occa is installed in another location outside of the build and source directories then this naming still make sense.